plutus-benchmark/flat/Main.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE BangPatterns      #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Control.DeepSeq              (force)
+import           Criterion.Main
+import           Criterion.Types
+import qualified Data.ByteString              as BS
+import           Data.Map.Strict              (Map, fromList, lookup)
+import           Data.Maybe                   (fromJust)
+import           Data.Text                    (Text, unpack)
+import           System.Mem                   (performMajorGC)
+
+import           Language.PlutusCore          (Name (..))
+import           Language.PlutusCore.Universe
+import           Language.UntypedPlutusCore
+
+import           Codec
+import           Dataset
+import           Report
+
+import           Prelude                      hiding (lookup)
+
+type Tm = Term Name DefaultUni ()
+
+serializedContracts :: Map (Text, Text) BS.ByteString
+serializedContracts = fromList $ do
+  (contractName, contract) <- contracts
+  (codecName, codec)       <- codecs
+  pure $ ((contractName, codecName), serialize codec contract)
+
+benchmarkSerialize ::
+     (Text, Tm)
+  -> (Text, Codec Tm)
+  -> Benchmark
+benchmarkSerialize (contractName, contract) (codecName, (Codec serialize' _)) =
+  bench (unpack $ " " <> contractName <> " - " <> codecName)
+        -- (nf (BS.length . serialize) contract) ~ original code
+        (nf serialize' contract)
+
+benchmarkDeserialize ::
+     Map (Text, Text) BS.ByteString
+  -> (Text, Tm)
+  -> (Text, Codec Tm)
+  -> Benchmark
+benchmarkDeserialize scs (contractName, _) (codecName, codec) =
+  let !serialized = fromJust $ lookup (contractName, codecName) scs
```

Comment:
comment needed!

Reply:
FWIW I think it would be fine to just do the serialization here so long as we force it. It's still outside the `bench` call.

Reply:
I did that initially but then noticed that I needed `scs` anyway.